test/Algebra/Graph/Test/Label.hs

```diff
@@ -0,0 +1,147 @@
+{-# LANGUAGE OverloadedLists #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Test.Label
+-- License    : MIT (see the file LICENSE)
+-- Stability  : experimental
+--
+-- Testsuite for "Algebra.Graph.Label".
+-----------------------------------------------------------------------------
+module Algebra.Graph.Test.Label (
+  -- * Testsuite
+  testLabel
+  ) where
+
+import Algebra.Graph.Test hiding (NonNegative)
+import Algebra.Graph.Test.Generic (size10)
+import Algebra.Graph.Label
+import Data.Monoid
+
+type UnaryFn a = a -> a
+type BinaryFn a = a -> a -> a
+type Plus a = BinaryFn a
+type Mult a = BinaryFn a
+type Star a = UnaryFn a
+type Identity a = a
+type Zero a = a
+type One a = a
+type Annihilator a = a
+
+associative :: Eq a => BinaryFn a -> a -> a -> a -> Property
+associative f a b c = (a `f` b) `f` c == a `f` (b `f` c) // "Associative property"
+
+commutative :: Eq a => BinaryFn a -> a -> a -> Property
+commutative f a b = a `f` b == b `f` a // "Commutative property"
+
+idempotent :: Eq a => BinaryFn a -> a -> Property
+idempotent f a = a `f` a == a // "Idempotence property"
+
+annihilator :: Eq a => BinaryFn a -> Annihilator a -> a -> Property
+annihilator f z a = conjoin
+  [ a `f` z == z // "Annihilator property right"
```

Comment:
I would drop the word "property" from all `//` annotations, writing simply "Right annihilator" in such cases.

By the way, I think this particular property is usually referred to as "Right zero".